game/data/doctrine.py

```diff
@@ -10,11 +16,23 @@ class GroundUnitProcurementRatios:
     ratios: dict[UnitClass, float]
 
     def for_unit_class(self, unit_class: UnitClass) -> float:
+        if not self.ratios:
```

Comment:
Isn't this already covered by the `try` immediately after?

Reply:
I've reverted the change and the tests still pass so all good.